src/lib/middleware/response-time-metrics.ts

```diff
@@ -55,12 +55,17 @@ export function responseTimeMetrics(
         // when pathname is undefined use a fallback
         pathname = pathname ?? collapse(req.path);
         let appName: string | undefined;
+        let connectionId: string | undefined;
         if (
             !flagResolver.isEnabled('responseTimeWithAppNameKillSwitch') &&
             (instanceStatsService.getAppCountSnapshot('7d') ??
                 appNameReportingThreshold) < appNameReportingThreshold
         ) {
+            // TODO: upgrade to x-unleash-appname
```

Comment:
We could also give preference to the new header if it's there?

This isn't the right line to give the suggestion on, but like this:
```suggestion
             appName = req.headers['x-unleash-appname'] ?? req.headers['unleash-appname'] ?? req.query.appName;
```

Reply:
Very good point. Incorporating the suggestion